packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -164,6 +177,34 @@ function UnconnectedNavigatorProvider(
 		} );
 	}, [] );
 
+	const goToParent: NavigatorContextType[ 'goToParent' ] =
+		useCallback( () => {
+			const currentPath =
+				currentLocationHistory.current[
+					currentLocationHistory.current.length - 1
+				].path;
+			if ( currentPath === undefined ) {
+				return;
+			}
+			const parentPath = findParent(
+				currentPath,
+				currentScreens.current
+			);
+			if ( parentPath === undefined ) {
+				return;
+			}
+			const isBack =
+				currentLocationHistory.current.length > 1 &&
+				currentLocationHistory.current[
+					currentLocationHistory.current.length - 2
+				].path === parentPath;
+			if ( isBack ) {
+				goBack();
```

Comment:
> This is clever! I just wonder if it's a good idea to "mix and match" goBack and goTo, ie. chronological and hierarchical paradigms. What do you think?

what would you do instead, we need to apply the exact same logic in this case.

Reply:
I thought more about it and I think that it should ok the way it is in this PR (I in a [different convo](https://github.com/WordPress/gutenberg/pull/47883#discussion_r1101839766) I actually proposed to move this "short circuit"  directly to the `goTo` function, so that it can be leveraged also when calling `goTo( '/path', {isBack: true})`